spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/blocking/client/BlockingLoadBalancerClient.java

```diff
@@ -89,25 +93,48 @@ public BlockingLoadBalancerClient(LoadBalancerClientFactory loadBalancerClientFa
 				.getSupportedLifecycleProcessors(
 						loadBalancerClientFactory.getInstances(serviceId, LoadBalancerLifecycle.class),
 						DefaultRequestContext.class, Object.class, ServiceInstance.class);
+		Request lbRequest = request instanceof Request ? (Request) request : new DefaultRequest<>();
+		if (lbRequest.getContext() instanceof TimedRequestContext) {
```

Comment:
why the special case? Can't this happen in a lifecycle method?

Reply:
Yes, it should. I must have mixed it up during refactoring - will remove.